drivers/modem/gsm_ppp.c

```diff
@@ -1113,7 +1113,7 @@ static int gsm_init(const struct device *dev)
 	gsm->context.data_imsi = minfo.mdm_imsi;
```

Comment:
Overall looks good, just minor nit about commit subject. As this change is for multiple modem drivers, better subject would be `drivers: modem: .....`

Reply:
Originally it had been done for gsm ppp only but it turned out that the rest of the drivers also have to be fixed. I've fixed the commit subject.